python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -1161,8 +1160,7 @@ class RandomForestClassifier(Base):
         -----------
         params : dict of new params
         """
-        # Resetting handle as __setstate__ overwrites with handle=None
-        self.handle.__setstate__(self.n_streams)
+        self.handle = Handle(self.n_streams)
```

Comment:
why is the handle getting deleted on the setstate when the handle is now pickleable? That might be a better solution